frontend/src/features/detailpage/DetailPageContent.tsx

```diff
@@ -0,0 +1,184 @@
+            Avbryt
+          </Button>
+        </Modal.Footer>
+      </Modal>
+      <Link
+        as={RouterLink}
+        to={'/' + AuthenticationPath.Auth + '/' + AuthenticationPath.Overview}
+        className={classes.backLink}
+      >
+        <ArrowLeftIcon />
+        Tilbake til oversikt
+      </Link>
+      <Heading level={2} size='medium'>
+        {systemUser.integrationTitle}
+      </Heading>
+      <Paragraph spacing>{systemUser.productName.toUpperCase()}</Paragraph>
+      <Textfield
+        label='Endre navn på systembruker'
+        className={classes.nameField}
+        size='small'
+        value={name}
+        onChange={(event) => setName(event.target.value)}
+      />
+      <div className={classes.rightsHeader}>
+        <Heading level={3} size='xxsmall' spacing>
+          Redigere rettigheter:
+        </Heading>
+        <Button variant='tertiary'>
+          <PlusCircleIcon />
+          Legg til rettighet
+        </Button>
+      </div>
+      {selectedRights.length === 0 && <div>Systembrukeren har ingen rettigheter</div>}
+      {selectedRights.map((right) => {
+        return (
+          <ActionBar
+            key={right.right}
+            title={right.right}
+            subtitle={right.serviceProvider}
+            color='neutral'
+            actions={
+              <Button
+                variant='tertiary'
+                size='small'
+                onClick={() => {
+                  setSelectedRights((old) =>
+                    old.map((selectedRight) =>
+                      selectedRight.right == right.right
+                        ? { ...selectedRight, deleted: !selectedRight.deleted }
+                        : selectedRight,
+                    ),
+                  );
+                }}
+              >
+                {right.deleted ? (
+                  <>
+                    Angre fjerning
+                    <ArrowUndoIcon />
+                  </>
+                ) : (
+                  <>
+                    Fjern rettighet
+                    <MinusCircleIcon />
+                  </>
+                )}
+              </Button>
+            }
+          >
+            {right.actions && (
+              <div>
+                <Paragraph size='small' spacing>
+                  Eventuell tekst om rettighetene kommer her.
+                </Paragraph>
+                <Paragraph size='small' spacing>
+                  Dersom du ønsker å begrense tilgangen som blir gitt, kan du klikke på knappene
+                  nedenfor for å fjerne de.
+                </Paragraph>
+                <Chip.Group size='small'>
+                  {right.actions.map((action) => {
+                    return (
+                      <Chip.Toggle
+                        key={action.name}
+                        selected={action.on}
+                        onClick={() => {
+                          setSelectedRights((old) => {
```

Comment:
This inner method here is incredibly hard to read. I recommend creating a utility method, so you can do `setSelectedRights(old => toggleRight(old, right, action)`.